WordPress/gutenberg
# Table Block: Add toolbar button to add a caption

Related to:

- #44965
- #45112
- #47325

## What?

This PR adds a toggle toolbar item on the table block toolbar to add/remove a caption.

## How?
~~The approach is the same as the other PRs, but I had a little trouble with the order of the caption buttons.~~

Update: I learned that a reusable caption component was added in #56606. Therefore, I have updated this PR to using this component as well.

## Testing Instructions

- Insert a table block.
- Confirm that an empty caption area doesn't appear when the block is selected.
- Toggle caption area from the block toolbar and confrim that the caption area gets focus.
- After entering the caption, confirm that the caption text is maintained when the focus is removed from the block.
- After deleting the caption text, confirm that the caption area disappears when the focus is removed from the block.
- Confirm that the caption text reappears when the undo operation is performed.
